display_list/dl_paint.h

```diff
@@ -66,6 +66,20 @@ class DlPaint {
     return *this;
   }
 
+  // In the Impeller backend, we'll *always* apply dithering to gradients, and
+  // *never* apply dithering otherwise, and there will be no user-facing feature
+  // to change this behavior.
+  //
+  // To temporarily match this behavior in the Skia backend, we tag gradients
+  // with a special flag, and then check for that flag when converting from
+  // DlPaint to SkPaint.
+  //
+  // TODO(matanl): Remove this flag when the Skia backend is removed,
+  // https://github.com/flutter/flutter/issues/112498.
+  bool isDitherHintForSkBackend() const {
```

Comment:
Simpler name? "requiresDithering"? "shouldBeDithered"? Not clear on why we need to name a backend for this property. It's true that only the Skia backend will need to do something with this information, but it's not like the information is only valid under Skia - it's just a new fundamental implicit property of a dither.

Reply:
Or, "isGradient" and leave it at that. Skia knows what it must do in that case. Impeller does it anyway.

Reply:
Fair enough! Done and updated.